### PR TITLE
fix:retained input reference on x icon click

### PR DESCRIPTION
**Pull request by saiprabhu-dandanayak**
### Fixes : [#18536](https://github.com/appsmithorg/appsmith/issues/18536)

I have raised this pr in-order to retain the cursor position inside the search field when clicking on the `x` icon .

### Table
Screenshot from 2024-09-12 13-56-18

### SingleSelectTree
Screenshot from 2024-09-12 13-57-21

### MultiTreeSelect
image

### MultiSelect
image

 

## Summary by CodeRabbit

- **New Features**
	- Enhanced search and filter clearing functionality across multiple components to automatically focus the input field after clearing, improving user experience.
  
- **Bug Fixes**
	- Added focus actions to clear buttons in various components to ensure immediate interaction with the input field post-clear.

- **Documentation**
	- Added console log statements for better tracking of query changes and dropdown opening events in the SelectComponent.

### CODE REVIEW

**Comment by saiprabhu-dandanayak** [comment]
**@appsmithorg/query-js-pod** , Please add reviewer for this pr if you have enough bandwidth.
Thanks in advance.

**Comment by NilanshBansal** [comment]
Assigned @rahulbarwal @jacquesikot from Widgets team for review. cc: @carinanfonseca

**Comment by saiprabhu-dandanayak** [comment]
Hi **@jacquesikot** , could you pls review this pr , pls let me know any changes needs to be done.
Thanks in advance.

**Comment by saiprabhu-dandanayak** [comment]
Hi **@jacquesikot** , could you pls review this pr , pls let me know any changes needs to be done. dute to inactivity pr got stale labelled.
Thanks in advance.

File: app/client/src/widgets/SelectWidget/component/index.tsx [hunk 1]
```diff
@@ -130,6 +130,7 @@ class SelectComponent extends React.Component<
   };
 
   onQueryChange = debounce((filterValue: string) => {
+    console.log(">> inside query change")
```
Comment (rahulbarwal) [inline app/client/src/widgets/SelectWidget/component/index.tsx:133]: Can be removed.

File: app/client/src/widgets/SelectWidget/component/index.tsx [hunk 2]
```diff
@@ -409,6 +410,7 @@ class SelectComponent extends React.Component<
               onClose: this.handleCloseList,
               // onActiveItemChange is called twice abd puts the focus on the first item https://github.com/palantir/blueprint/issues/4192
               onOpening: () => {
+                console.log(">>> opend...")
```
Comment (rahulbarwal) [inline app/client/src/widgets/SelectWidget/component/index.tsx:413]: can be removed.

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Missing tests. Please add tests(unit, preferably) for these changes.

File: app/client/src/widgets/SelectWidget/component/index.tsx (hunk 1 above)
Comment (saiprabhu-dandanayak) [reply app/client/src/widgets/SelectWidget/component/index.tsx]: removed

File: app/client/packages/design-system/widgets-old/package.json [hunk 3]
```diff
@@ -7,11 +7,16 @@
   "license": "Apache-2.0",
   "scripts": {
     "lint": "yarn g:lint",
-    "prettier": "yarn g:prettier"
+    "prettier": "yarn g:prettier",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:coverage": "jest --coverage"
```
Comment (rahulbarwal) [inline app/client/packages/design-system/widgets-old/package.json:13]: Do we need all these "scripts"?

Comment (saiprabhu-dandanayak) [reply app/client/packages/design-system/widgets-old/package.json:13]: Actually these were not there in `design-system` so in-order to test the file i have added respective packages.

**Comment by rahulbarwal** [comment]
@saiprabhu-dandanayak quality checks are failing, please fix them: https://github.com/appsmithorg/appsmith/actions/runs/11046247172/job/30685240765?pr=36553

You can run the failing commands on your local to confirm if the issue has been fixed.

**Review by rahulbarwal — COMMENTED** [review]
CI is still failing 😞 : https://github.com/appsmithorg/appsmith/actions/runs/11065536238/job/30745146672?pr=36553
Please check. 

One more thing, lets not name our test file `index.test.tsx` : this does not tell me anything about the tests.

**Comment by rahulbarwal** [comment]
Hey @saiprabhu-dandanayak, All three commands are failing on your latest changes: 
* yarn run lint:ci 
* yarn run test:unit:ci
* yarn run check-types

here is a sample run: https://app.warp.dev/block/Hwm6Md43BhO2h02LSBQTx8

Can you try all three commands in your local and get back when they succed?

**Comment by jacquesikot** [comment]
Hey @saiprabhu-dandanayak are you still working on this PR - have you been able to address the issues @rahulbarwal mentioned?

**Pull request closed without merging**